includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -304,27 +310,33 @@ public function process_payment( $order_id ) {
 		$order               = wc_get_order( $order_id );
 		$amount              = $order->get_total();
 		$currency            = $order->get_currency();
+		$payment_needed      = true;
 		$save_payment_method = ! empty( $_POST[ 'wc-' . static::GATEWAY_ID . '-new-payment-method' ] ); // phpcs:ignore WordPress.Security.NonceVerification.Missing
 		$token               = Payment_Information::get_token_from_request( $_POST ); // phpcs:ignore WordPress.Security.NonceVerification.Missing
 
 		if ( $payment_intent_id ) {
 			list( $user, $customer_id ) = $this->manage_customer_details_for_order( $order );
 
-			$this->payments_api_client->update_intention(
-				$payment_intent_id,
-				WC_Payments_Utils::prepare_amount( $amount, $currency ),
-				strtolower( $currency ),
-				$save_payment_method,
-				$customer_id,
-				$this->get_level3_data_from_order( $this->account->get_account_country(), $order )
-			);
+			$converted_amount = WC_Payments_Utils::prepare_amount( $amount, $currency );
+			$payment_needed   = 0 < $converted_amount;
```

Comment:
Would we be able to move the `converted_amount` and `payment_needed` assignments to be above the if statement? I only ask because having just done the block checkout work, there is a scenario where the `$payment_intent_id` isn't passed in. If that scenario happens with block checkout and a free trial subscription, the flow would error out as the client side would try to create a payment intent instead of a setup intent. I think moving those variable assignments above and outside of the conditional would prevent that issue without changing much.

Reply:
Cool, sounds good to me. I actually think is a better solution than just blindly setting `payment_needed` as `true` by default. This should now be updated in the latest changes I've just pushed.